### PR TITLE
Small fixes

### PR DESCRIPTION
**Pull request by abergeron**
# Description
This is a collection of small fixes for bugs I found while investigating the code.

If one of the fixes is questionable, I can remove the commit in question from the PR on request.

# Checklist

## Documentation
- [x] I have updated the relevant documentation related to my changes

## Quality
- [x] I have read the [CONTRIBUTING](https://github.com/Epistimio/orion/blob/develop/CONTRIBUTING.md) doc
- [x] My commits messages follow [this format](https://chris.beams.io/posts/git-commit/)
- [x] My code follows the style guidelines (`$ tox -e lint`)

### CODE REVIEW

File: src/orion/client/__init__.py [hunk 1]
```diff
@@ -338,9 +338,7 @@ def workon(
             max_broken=max_broken,
         )
 
-        producer = Producer(experiment)
-
-        experiment_client = ExperimentClient(experiment, producer)
+        experiment_client = ExperimentClient(experiment)
```
Comment (lebrice) [inline src/orion/client/__init__.py:341]: Yes exactly! I found the same bug thanks to type hints on my end yesterday:
potential_bug

**Comment by abergeron** [comment]
The pre-commit hook complains about things that did not change in this PR.  Should I just ignore it?

**Comment by bouthilx** [comment]
It's weird that these errors do not show up in the CI for the dev branch. If you are able to fix these it would be great. Otherwise we'll do it in another PR.

**Comment by abergeron** [comment]
I did a close/reopen to re-run the tests

**Comment by abergeron** [comment]
The current failure doesn't happen locally for me and I have no idea why it does happen.

**Pull request closed without merging**